androidApp/src/androidTest/kotlin/com/sagebionetworks/assessmentmodel/NodeNavigatorUITests.kt

```diff
@@ -0,0 +1,182 @@
+        composeTestRule.onNodeWithText("Review Instructions").performClick()
+        assertTrue(assessmentViewModel.showFullInstructions)
+
+        navigateForward("Start", 1)
+        navigateForward("Next", 3)
+        navigateForward("Skip question", 4)
+
+        insertIntoExpected(
+            assessmentViewModel.showFullInstructions,
+            surveyAExpectedStepHistory,
+            listOf("completion", "multipleChoice", "favoriteFood", "followupQ", "choiceQ1", "step2", "step1", "addedStep1", "overview"),
+            4
+        )
+        compareStepHistory(actualStepHistory, surveyAExpectedStepHistory)
+    }
+
+
+    private fun passThroughSampleAssessment(showFullInstructions: Boolean) {
+        onView(withText("sampleId")).perform(click())
+        setupActivityAndViewModel(showFullInstructions)
+
+        navigateForward("Next", if (assessmentViewModel.showFullInstructions) 2 else 1)
+        navigateForward("Skip question", 7)
+
+        val actualStepHistory = assessmentViewModel.currentResult.pathHistoryResults
+        insertIntoExpected(assessmentViewModel.showFullInstructions, sampleExpectedStepHistory, listOf("step2"), 1)
+        compareStepHistory(actualStepHistory, sampleExpectedStepHistory)
+
+        navigateForward("Exit", 1)
+    }
+
+    private fun passThroughWrapperAssessment(showFullInstructions: Boolean) {
+        onView(withText("sampleWrapperId")).perform(click())
+        setupActivityAndViewModel(showFullInstructions)
+
+        navigateForward("Next", 1)
+
+        // Use the child assessment's view model for this wrapped assessment case
+        assessmentViewModel = assessmentViewModel.currentChild as BranchNodeState
+        assessmentViewModel.showFullInstructions = showFullInstructions
+
+        navigateForward("Next", if (assessmentViewModel.showFullInstructions) 2  else 1)
+        navigateForward("Skip question", 7)
+
+        val actualStepHistory = assessmentViewModel.currentResult.pathHistoryResults
+        insertIntoExpected(assessmentViewModel.showFullInstructions, sampleExpectedStepHistory, listOf("step2"), 1)
+        compareStepHistory(actualStepHistory, sampleExpectedStepHistory)
+
+        navigateForward("Exit", 1)
+    }
+
+    private fun passThroughSurveyA(showFullInstructions: Boolean) {
+        onView(withText("survey_a")).perform(click())
+
+        setupActivityAndViewModel(showFullInstructions)
+
+        navigateForward("Start", 1)
+        navigateForward("Next", if (assessmentViewModel.showFullInstructions) 3 else 0)
+        navigateForward("Skip question", 4)
+
+        val actualStepHistory = assessmentViewModel.currentResult.pathHistoryResults
+        insertIntoExpected(assessmentViewModel.showFullInstructions, surveyAExpectedStepHistory, listOf("step2", "step1", "addedStep1"), 1)
+        compareStepHistory(actualStepHistory, surveyAExpectedStepHistory)
+
+        navigateForward("Exit", 1)
```

Comment:
What happens in the UI tests if the locale of the test device is not English?

Reply:
Gotcha, localized strings in the NodeNavigatorUITests in latest commit.